src/errors.ts

```diff
@@ -0,0 +1,68 @@
+
+// Error codes indicating the reson for an error.
+export enum PurchasesErrorCode {
+    UnknownError = 0,
+    PurchaseCancelledError = 1,
+    StoreProblemError = 2,
+    PurchaseNotAllowedError = 3,
+    PurchaseInvalidError = 4,
+    ProductNotAvailableForPurchaseError = 5,
+    ProductAlreadyPurchasedError = 6,
+    ReceiptAlreadyInUseError = 7,
+    InvalidReceiptError = 8,
+    MissingReceiptFileError = 9,
+    NetworkError = 10,
+    InvalidCredentialsError = 11,
+    UnexpectedBackendResponseError = 12,
+    ReceiptInUseByOtherSubscriberError = 13,
+    InvalidAppUserIdError = 14,
+    OperationAlreadyInProgressError = 15,
+    UnknownBackendError = 16,
+    InvalidAppleSubscriptionKeyError = 17,
+    IneligibleError = 18,
+    InsufficientPermissionsError = 19,
+    PaymentPendingError = 20,
+    InvalidSubscriberAttributesError = 21,
+    LogOutAnonymousUserError = 22,
+}
+
+export interface PurchasesError {
+    code: string;
```

Comment:
This was a `number` before, doesn't it come as an int from purchases hybrid common https://github.com/RevenueCat/purchases-hybrid-common/blob/develop/ios/PurchasesHybridCommon/PurchasesHybridCommon/RCErrorContainer.m#L11?

Reply:
that's what I thought initially too. However [this comment](https://github.com/RevenueCat/react-native-purchases/issues/228#issuecomment-818753249) made me dig deeper and it turns out we were transforming the code into a string: 

https://github.com/RevenueCat/react-native-purchases/blob/develop/ios/RNPurchases.m#L346
https://github.com/RevenueCat/react-native-purchases/blob/develop/android/src/main/java/com/revenuecat/purchases/react/RNPurchasesModule.java#L115

We're even checking against it as a string: https://github.com/RevenueCat/react-native-purchases/blob/develop/src/index.ts#L803

So it's a string that will always hold a number. 

I didn't want to change the existing behavior so as not to break any error handling from customers. And hopefully from now on, folks will use the new enum for error codes.

Reply:
If we make PurchasesErrorCodes a string enum, can we change this "code" to be a PurchasesErrorCode instead of String. Would that remove the need to call getErrorCode when trying to get the code?

Reply:
great idea! I'll give it a shot

Reply:
Thanks! I am not sure if it's possible but hopefully would make it easier at least for people using Typescript